linera-core/src/client/mod.rs

```diff
@@ -2711,9 +2714,12 @@ where
         &self,
         bytes: Vec<Vec<u8>>,
     ) -> Result<ClientOutcome<Certificate>, ChainClientError> {
-        let blobs = bytes.into_iter().map(Blob::new_data);
+        let mut blobs = Vec::with_capacity(bytes.len());
+        for byte_vec in bytes {
+            blobs.push(Blob::new_data(byte_vec));
+        }
```

Comment:
(Maybe more concise?)

```suggestion
        let blobs = bytes.into_iter().map(Blob::new_data).collect::<Vec<_>>();
```